exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricTranslator.java

```diff
@@ -0,0 +1,213 @@
+        return value.getStringValue();
+      case LONG:
+        return value.getLongValue() + "";
+      case DOUBLE:
+        return value.getDoubleValue() + "";
+      case BOOLEAN:
+        return value.getBooleanValue() + "";
+      case STRING_ARRAY:
+        return String.join(", ", value.getStringArrayValue());
+      case LONG_ARRAY:
+        return value.getLongArrayValue().stream().map(Object::toString)
+            .collect(Collectors.joining(", "));
+      case DOUBLE_ARRAY:
+        return value.getDoubleArrayValue().stream().map(Object::toString)
+            .collect(Collectors.joining(", "));
+      case BOOLEAN_ARRAY:
+        return value.getBooleanArrayValue().stream().map(Object::toString)
+            .collect(Collectors.joining(", "));
+      default:
+        return null;
+    }
+  }
+
+  static LabelDescriptor mapConstantLabel(String key, String value) {
+    LabelDescriptor.Builder builder = LabelDescriptor.newBuilder().setKey(key);
+    if ("true".equalsIgnoreCase(value) || "false".equalsIgnoreCase(value)) {
+      builder.setValueType(LabelDescriptor.ValueType.BOOL);
+    } else if (Longs.tryParse(value) != null) {
+      builder.setValueType(LabelDescriptor.ValueType.INT64);
+    } else {
+      builder.setValueType(LabelDescriptor.ValueType.STRING);
+    }
+    return builder.build();
+  }
+
+  static Point mapPoint(Map<MetricWithLabels, Long> lastUpdatedTime, MetricData metric, MetricWithLabels updateKey,
+      Instant exporterStartTime, long pointCollectionTime) {
+    Builder pointBuilder = Point.newBuilder();
+    Type type = metric.getDescriptor().getType();
+    if (LONG_TYPES.contains(type)) {
+      pointBuilder.setValue(TypedValue.newBuilder().setInt64Value(
+          ((MetricData.LongPoint) metric.getPoints().iterator().next()).getValue()));
+    } else if (DOUBLE_TYPES.contains(type)) {
+      pointBuilder.setValue(TypedValue.newBuilder().setDoubleValue(
+          ((MetricData.DoublePoint) metric.getPoints().iterator().next()).getValue()));
+    } else {
+      logger.log(WARNING, "Type {0} not supported", type);
+      return null;
+    }
+    pointBuilder.setInterval(
+        mapInterval(lastUpdatedTime, updateKey, type, exporterStartTime, pointCollectionTime));
+    return pointBuilder.build();
+  }
+
+  static TimeInterval mapInterval(Map<MetricWithLabels, Long> lastUpdatedTime,
+      MetricWithLabels updateKey, Type descriptorType, Instant exporterStartTime, long pointCollectionTime) {
+    long seconds;
+    int nanos;
+    if (CUMULATIVE_TYPES.contains(descriptorType)) {
+      if (!lastUpdatedTime.containsKey(updateKey)) {
+        // The aggregation has not reset since the exporter
+        // has started up, so that is the start time
+        seconds = exporterStartTime.getEpochSecond();
+        nanos = exporterStartTime.getNano();
+      } else {
+        // The aggregation reset the last time it was exported
+        // Add 1ms to guarantee there is no overlap from the previous export
```

Comment:
This seems artificial / misleading, but fine to discuss later if there's a better approach.